src/main/java/com/faforever/api/leaderboard/LeaderboardController.java

```diff
@@ -32,12 +35,22 @@ public LeaderboardController(LeaderboardService leaderboardService) {
     this.leaderboardService = leaderboardService;
   }
 
+
   @Async
   @RequestMapping(path = "/ladder1v1", method = RequestMethod.GET)
   @ApiOperation("Lists the ladder1v1 leaderboard")
-  public CompletableFuture<JsonApiDocument> getLadder1v1(@RequestParam(value = "page[number]", required = false) Integer page,
-                                                         @RequestParam(value = "page[size]", required = false) Integer pageSize) {
-    List<Resource> values = StreamSupport.stream(leaderboardService.getLadder1v1Leaderboard(page, pageSize).spliterator(), false)
+  public CompletableFuture<JsonApiDocument> getLadder(@RequestParam(value = "page[number]", required = false) Integer page,
+                                                      @RequestParam(value = "page[size]", required = false) Integer pageSize,
+                                                      @ApiParam("Only players that have the following pattern in their name.")
+                                                      @RequestParam(value = "playerNameMatchesRegex", required = false) String playerNameMatchesRegex) {
```

Comment:
playerNameMatchesRegex is very confusing/wrong name, it's not regex and Matches suggests its boolean.
I would leave just playerName or playerNamePattern

Reply:
playerNamePattern it is